### PR TITLE
Bui

### PR DESCRIPTION
**Pull request by UAChronos**

### CODE REVIEW

File: RPGSystem/ConsumablesConfigurator.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+        }
+
+        #endregion
+
+        private System.Windows.Forms.TableLayoutPanel tableLayoutPanel1;
+        private System.Windows.Forms.Label lblCraftingPieces;
+        private System.Windows.Forms.Label lblGold;
+        private System.Windows.Forms.TextBox txtAmmountOfGold;
+        private System.Windows.Forms.TextBox txtAmmountOfCraftingPieces;
```
Comment (Krum78) [inline RPGSystem/ConsumablesConfigurator.Designer.cs]: Change to numeric editor

File: RPGSystem/ConsumablesConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            Close();
+        }
+
+        private void ConsumablesConfigurator_Load(object sender, EventArgs e)
+        {
+            Gold = GameDataService.Instance.GameModel.Gold;
+            CraftingPieces = GameDataService.Instance.GameModel.CraftingPieces;
+
+            txtAmmountOfGold.Text = Gold.ToString();
+            txtAmmountOfCraftingPieces.Text = CraftingPieces.ToString();
```
Comment (Krum78) [inline RPGSystem/ConsumablesConfigurator.cs]: When you change the control type there will be no need for string conversion and parsing

File: RPGSystem/DiceTool.Designer.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            this.Text = "Dice";
+            this.tableLayoutPanel1.ResumeLayout(false);
+            this.tableLayoutPanel1.PerformLayout();
+            this.ResumeLayout(false);
+
+        }
+
+        #endregion
+
+        private System.Windows.Forms.TableLayoutPanel tableLayoutPanel1;
```
Comment (Krum78) [inline RPGSystem/DiceTool.Designer.cs:109]: Name the layout panel to reflect it's purpose

File: RPGSystem/Form1.Designer.cs [hunk 4]
```diff
@@ -29,12 +29,222 @@ protected override void Dispose(bool disposing)
         private void InitializeComponent()
         {
             this.components = new System.ComponentModel.Container();
+            this.menuStrip1 = new System.Windows.Forms.MenuStrip();
```
Comment (Krum78) [inline RPGSystem/Form1.Designer.cs]: Names of controls or components used in the application should describe purpose of them.

File: RPGSystem/Form1.Designer.cs [hunk 5]
```diff
@@ -29,12 +29,222 @@ protected override void Dispose(bool disposing)
...
+            this.saveAsToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.openToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.gameToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.addPlayerToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.diceToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.editToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.lockToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.contextMenuStrip1 = new System.Windows.Forms.ContextMenuStrip(this.components);
+            this.flowLayoutPanel1 = new System.Windows.Forms.FlowLayoutPanel();
+            this.flowLayoutPanel2 = new System.Windows.Forms.FlowLayoutPanel();
```
Comment (Krum78) [inline RPGSystem/Form1.Designer.cs]: The same naming problem

File: RPGSystem/Form1.cs [hunk 6]
```diff
@@ -16,5 +16,17 @@ public Form1()
         {
```
Comment (Krum78) [inline RPGSystem/Form1.cs:20]: Please rename Form1 to something meaningful

File: RPGSystem/GameForm.Designer.cs [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        #region Windows Form Designer generated code
+
+        /// <summary>
+        /// Required method for Designer support - do not modify
+        /// the contents of this method with the code editor.
+        /// </summary>
+        private void InitializeComponent()
+        {
+            this.components = new System.ComponentModel.Container();
+            this.menuStrip1 = new System.Windows.Forms.MenuStrip();
```
Comment (Krum78) [inline RPGSystem/GameForm.Designer.cs:32]: The same problem - naming

File: RPGSystem/GameForm.Designer.cs [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+            this.saveAsToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.openToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.gameToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.addPlayerToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.diceToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.editToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.lockToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.contextMenuStrip1 = new System.Windows.Forms.ContextMenuStrip(this.components);
+            this.flowLayoutPanel1 = new System.Windows.Forms.FlowLayoutPanel();
+            this.flowLayoutPanel2 = new System.Windows.Forms.FlowLayoutPanel();
```
Comment (Krum78) [inline RPGSystem/GameForm.Designer.cs:44]: The same problem - naming

File: RPGSystem/PlayerStats.Designer.cs [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+        #region Component Designer generated code
+
+        /// <summary> 
+        /// Required method for Designer support - do not modify 
+        /// the contents of this method with the code editor.
+        /// </summary>
+        private void InitializeComponent()
+        {
+            this.components = new System.ComponentModel.Container();
+            this.contextMenuStrip1 = new System.Windows.Forms.ContextMenuStrip(this.components);
```
Comment (Krum78) [inline RPGSystem/PlayerStats.Designer.cs:32]: The same problem - naming

File: RPGSystem/PlayerStats.Designer.cs [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+            this.lblPlayersClass.Font = new System.Drawing.Font("Monotype Corsiva", 15.75F, ((System.Drawing.FontStyle)((System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic))));
+            this.lblPlayersClass.Location = new System.Drawing.Point(3, 43);
+            this.lblPlayersClass.Name = "lblPlayersClass";
+            this.lblPlayersClass.Size = new System.Drawing.Size(65, 25);
+            this.lblPlayersClass.TabIndex = 7;
+            this.lblPlayersClass.Text = "Class:";
+            // 
+            // label1
+            // 
+            this.label1.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right)));
```
Comment (Krum78) [inline RPGSystem/PlayerStats.Designer.cs:126]: The same problem - naming

File: RPGSystem/Program.cs [hunk 11]
```diff
@@ -16,7 +16,7 @@ static void Main()
         {
             Application.EnableVisualStyles();
             Application.SetCompatibleTextRenderingDefault(false);
-            Application.Run(new Form1());
+            Application.Run(new GameForm());
```
Comment (Krum78) [inline RPGSystem/Program.cs:19]: Looks like the Form1 source code is a leftover after improperly made renaming refactoring.

File: RPGSystem/RandomNumberGenerator.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RPGSystem
+{
+    class RandomNumberGenerator
```
Comment (Krum78) [inline RPGSystem/RandomNumberGenerator.cs:12]: Put summary descriptions for all classes and their public, protected and internal members (properties, functions, constructors, etc.)
Make the same changes in all classes (excluding generated code)

File: RPGSystem/DiceTool.cs [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    public partial class DiceTool : Form
+    {
+        public DiceTool()
+        {
+            InitializeComponent();
+        }
+
+        private void BtnThrewADice_Click(object sender, EventArgs e)
+        {
+            RandomNumberGenerator randomNumberGenerator = new RandomNumberGenerator();
```
Comment (Krum78) [inline RPGSystem/DiceTool.cs]: I would recommend making randomNumberGenerator private class field and initialize it inline to not to create a new instance every time a user clicks on a throw button.

**Review by Krum78 — COMMENTED** [review]
Needs work

File: RPGSystem/ConsumablesConfigurator.cs (hunk 2 above)
Comment (UAChronos) [reply RPGSystem/ConsumablesConfigurator.cs]: Fixed

File: RPGSystem/DiceTool.cs (hunk 13 above)
Comment (UAChronos) [reply RPGSystem/DiceTool.cs]: Fixed

File: RPGSystem/Form1.cs (hunk 6 above)
Comment (UAChronos) [reply RPGSystem/Form1.cs:20]: File was deleted

File: RPGSystem/Program.cs (hunk 11 above)
Comment (UAChronos) [reply RPGSystem/Program.cs:19]: Leftovers were deleted

**Review by UAChronos — COMMENTED** [review]
1

**Pull request merged**